airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/OssMetricsRegistry.java

```diff
@@ -137,21 +139,53 @@ public enum OssMetricsRegistry implements MetricsRegistry {
       "number of records synced during replication"),
   RESET_REQUEST(MetricEmittingApps.WORKER,
       "reset_request",
-      "number of requested resets");
+      "number of requested resets"),
+
+  ATTEMPT_CREATED(
+      MetricEmittingApps.WORKER,
+      "attempt_created",
+      "increments when a new attempt is created. one is emitted per attempt",
+      MetricTags.DATA_PLANE_ID,
+      MetricTags.MIN_CONNECTOR_RELEASE_STATE, // max
+      MetricTags.MAX_CONNECTOR_RELEASE_STATE // min
+  ),
+  // ATTEMPT_PENDING (doesn't actually mean anything)
+  ATTEMPTS_RUNNING(
+      MetricEmittingApps.METRICS_REPORTER,
+      "attempts_running",
+      "number attempts in status running at a moment in time.",
+      MetricTags.DATA_PLANE_ID, // really should be data-plane id
+      MetricTags.MIN_CONNECTOR_RELEASE_STATE, // max
+      MetricTags.MAX_CONNECTOR_RELEASE_STATE // min
+  ),
+  ATTEMPT_COMPLETED(
+      MetricEmittingApps.WORKER,
+      "attempt_completed",
+      "increments when a new attempt is completed. one is emitted per attempt",
+      MetricTags.DATA_PLANE_ID,
+      MetricTags.MIN_CONNECTOR_RELEASE_STATE, // max
+      MetricTags.MAX_CONNECTOR_RELEASE_STATE, // min
+      MetricTags.ATTEMPT_OUTCOME,
+      MetricTags.FAILURE_ORIGIN,
+      MetricTags.FAILURE_TYPE);
 
   private final MetricEmittingApp application;
   private final String metricName;
   private final String metricDescription;
 
+  private final List<String> metricTags;
+
   OssMetricsRegistry(final MetricEmittingApp application,
                      final String metricName,
-                     final String metricDescription) {
+                     final String metricDescription,
+                     final String... metricTags) {
```

Comment:
i like this idea even though we can't strictly enforce it. Perhaps we can do slightly better by turning this into a list of metric tags and having a static constant named `NO_METRIC_TAGS` to make it obvious.

Reply:
@davinchia That would add additional logic to check the list to see if it only has one entry and it's value is `NO_METRIC_TAGS`.  That seems more complicated than using the ellipses to indicate that the list/array can simply not be provided if there are no tags.  This also aligns with how the OpenTracing/DataDog tracing classes work.